DirectEmployers/MyJobs
# Stop premature queryset evaluation.

## What's this PR do?
Stops the premature evaluation of querysets.
Checking the truthiness of a queryset apparently forces evaluation of said queryset. Since, in the cases I have corrected, the queryset is mostly unfiltered, this causes some pretty large queries to be run. Needless to say, that's bad.

## How should this be manually tested?
Set up a site to use facet tags. Check # of queries before and after this branch. The number of queries will have gone down by 1. Trust me that this is a pretty big query.

## Any background context you want to provide?
Discovered while setting up a site to use the new faceting system. 

## PR Checklist

### Separation of Concerns
